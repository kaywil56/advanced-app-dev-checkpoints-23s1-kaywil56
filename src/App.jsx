import AddItem from "./components/AddItem";
import TodoList from "./components/TodoList";
import Login from "./components/Login";
import { getAuth, onAuthStateChanged, signOut } from "firebase/auth";
import { createUser } from "./firestore.service";
import { useEffect, useState } from "react";
import { Context } from "./context";
import UserInfo from "./components/UserInfo";

const App = () => {
  const auth = getAuth();
  const [authContext, setAuthContext] = useState({});
  const [unsubscribe, setUnsubscribe] = useState(null);

  const signOutUser = () => {
    unsubscribe();
    signOut(auth)
  };

  useEffect(() => {
    onAuthStateChanged(auth, (user) => {
      if (user) {
        setAuthContext(user);
        createUser(user);
      } else {
        setAuthContext({});
      }
    });
  }, []);

  return (
    <Context.Provider value={[authContext, setAuthContext]}>
      {authContext.uid ? (
        <main>
          <h1>Todo</h1>
          <AddItem />
          <TodoList setUnsubscribe={setUnsubscribe} />
          <button onClick={signOutUser} id="sign-out">
            Sign out
          </button>
          <UserInfo />
        </main>
      ) : (
        <Login />
      )}
    </Context.Provider>
  );
};

export default App;
