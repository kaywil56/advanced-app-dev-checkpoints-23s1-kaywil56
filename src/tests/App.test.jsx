import { render } from "@testing-library/react";
import { describe, test, expect } from "vitest";
import App from "../App";

describe("App", () => {
  test("Test app is rendered correctly", () => {
    const {asFragment, getByText} = render(<App />)
    expect(asFragment()).toMatchSnapshot()
  });
});
