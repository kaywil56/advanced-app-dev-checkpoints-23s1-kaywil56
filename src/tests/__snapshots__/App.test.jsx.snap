// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > Test app is rendered correctly 1`] = `
<DocumentFragment>
  <h1>
    Todo List
  </h1>
  <ul />
  <form>
    <input
      aria-label="title"
      name="title"
      placeholder="Title"
      type="text"
      value=""
    />
    <textarea
      aria-label="description"
      id="description"
      name="description"
      placeholder="Description"
      type="text"
    />
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</DocumentFragment>
`;
